src/codegen/function.rs

```diff
@@ -50,17 +52,17 @@ pub fn declaration(library: &library::Library, analysis: &analysis::functions::I
     format!("fn {}({}){}", analysis.name, param_str, return_str)
 }
 
-macro_rules! write_to_vec(
-    ($dst:expr, $($arg:tt)*) => (
-        $dst.push(fmt::format(format_args!($($arg)*)))
-    )
-);
+pub fn body(library: &library::Library, analysis: &analysis::functions::Info,
+    in_trait: bool) -> Vec<String> {
+    let mut builder = Builder::new()
+        .glib_name(&analysis.glib_name)
+        .from_glib(analysis.ret.to_return_value_body(&library, &analysis));

```

Comment:
Could you give a bit more thought to the names here? Wouldn't the `from_glib` and `to_return_value_body` methods have related names? Right now the only way to make sense of what happens here is to read a few function bodies.